tests/schema/test_docstring.py

```diff
@@ -0,0 +1,293 @@
+
+    Attributes:
+        fields: Names of the fields used as primary keys
+    """
+
+    fields: List[str]
+
+
+@strawberry.directive(locations=[DirectiveLocation.FIELD])
+def replace(value: str, old_char: str, new_char: str):
+    """
+    Replaces a character in the string
+
+    Args:
+        value: String being edited
+        old_char: Character being removed
+        new_char: Character being added
+    """
+    return value.replace(old_char, new_char)
+
+
+@strawberry.enum
+class EnumType(Enum):
+    """
+    Enum type description
+
+    Attributes:
+        VANILLA: Vanilla is a spice derived from orchids of the genus Vanilla
+        STRAWBERRY: The garden strawberry is a widely grown species of the genus Fragaria
+        CHOCOLATE: Chocolate is a food product made from roasted and ground cacao pods
+    """
+
+    VANILLA = "vanilla"
+    STRAWBERRY = "strawberry"
+    CHOCOLATE = "chocolate"
+    ROCKY_ROAD = "rocky_road"
+    """Chocolate ice cream with nuts and marshmallows"""
+
+
+@strawberry.interface
+class InterfaceType:
+    """
+    This is a graphQL interface type
+
+    Attributes:
+        var_a (int): A field that doesn't need a resolver
+        var_b (int): A complex field with a resolver
+    """
+
+    var_a: int
+
+    @strawberry.field
+    def var_b(self, documented: int = 1, undocumented: str = 2) -> str:
+        """
+        A complex field with a resolver and args
+
+        Args:
+            documented (int): Something
+        """
+        return ""
+
+
+@strawberry.type(directives=[Keys(fields=["var_f"])])
+class ObjectType(InterfaceType):
+    """
+    This is a graphQL object type
+
+    Attributes:
+        var_c (str): Something to differentiate ObjectType from InterfaceType
+        var_d:
+    """
+
+    var_c: EnumType
+    var_d: float  # Empty description is ignored
+    var_e: int  # No description
+    var_f: str
+    """ PEP 257 description """
+
+
+@strawberry.input
+class InputType:
+    """
+    This is a graphQL input type
+
+    Description has many lines.
+    Many, many lines
+
+    Attributes:
+        var_a (int): Something
+    """
+
+    var_a: int
+
+
+@strawberry.type
+class Query:
+    """
+    Main entrypoint to the GraphQL reads
+
+    Attributes:
+        object: A GraphQL object
+    """
+
+    object: ObjectType
+
+
+@strawberry.type
+class Mutation:
+    """
+    Main entrypoint to the GraphQL updates
+
+    Attributes:
+        set_object: A GraphQL object
+    """
+
+    @strawberry.field
+    def set_object(new_data: InputType, undocumented: str) -> str:
+        """
+        Update the object
+
+        Args:
+            new_data: New data
+            x-undocumented: This fields won't have a GraphQL description
+        """
+        return ""
+
+
+def test_docstrings_enabled():
+    schema = strawberry.Schema(
+        query=Query,
+        mutation=Mutation,
+        directives=[replace],
+        config=StrawberryConfig(description_sources=DescriptionSource),
```

Comment:
Why are you passing the enum _Type_ to `description_sources` instead of a list of instances?

Reply:
Potentially `DescriptionSource.__members__.values()` would be the most clear. It's the only option that doesn't rely on the `Enum.__iter__` magic, but I don't think `list(DescriptionSource)` is _bad_.

I'm actually surprised that passing the `DescriptionSource` class here (without converting it to a `list`) doesn't fail mypy checks, because everything is annotated as taking `List[DescriptionSource]` and not just `Iterable[DescriptionSource]`. I think we should make the `config=` above more explicit, but I also think the annotation for `description_sources` should be changed to `Iterable` everywhere. What do you think?